packages/vitest/src/node/config/resolveConfig.ts

```diff
@@ -798,7 +798,7 @@ export function resolveConfig(
 
   resolved.browser ??= {} as any
   resolved.browser.enabled ??= false
-  resolved.browser.headless ??= isCI
+  resolved.browser.headless ??= resolved.browser.enabled ? isCI : true
```

Comment:
Why does it matter if it's headless or now when `browser` is disabled?

Reply:
I think it's some other bug, but something keeps opening browser when this line is reverted. The `browser.ui` is `false` and `browser.enabled` is `false`.

Reply:
So does it still open them, but in headless mode? 🤔

Reply:
I don't know what opening in headless mode means. When I'm using `headless: false, ui: false`, no browsers are opened. But without this change, an actual browser opens on desktop.

Reply:
> I don't know what opening in headless mode means

It means the browser is opened in the background without a visual shell. I have a suspicious that this change just hides the issue because you can't see the browser window